### PR TITLE
#1095 - Tag Creator process for MCP (previously ACS Tools Tag Maker)

### PR DESCRIPTION
**Pull request by davidjgonzalez**

### CODE REVIEW

File: bundle/src/main/java/com/adobe/acs/commons/mcp/form/FieldComponent.java [hunk 1]
```diff
@@ -133,7 +133,7 @@ public String getName() {
 
     public boolean hasOption(String optionName) {
         return Stream.of(formField.options())
-                .filter(s -> s.equalsIgnoreCase(optionName))
+                .filter(s -> s.equalsIgnoreCase(optionName) || s.startsWith(optionName + "="))
```
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/mcp/form/FieldComponent.java]: @badvision does this look OK?

File: bundle/src/main/java/com/adobe/acs/commons/mcp/util/AnnotatedFieldDeserializer.java [hunk 2]
```diff
@@ -84,7 +86,13 @@ private static void parseInput(Object target, ValueMap input, Field field) throw
             if (value.getClass().isArray()) {
                 val = ((Object[]) value)[0];
             }
-            parseInputValue(target, String.valueOf(val), field);
+
+            if (val instanceof InputStream) {
```
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/mcp/util/AnnotatedFieldDeserializer.java]: @badvision Changed this to handle file uploads.

File: bundle/src/main/java/com/adobe/acs/commons/mcp/impl/ControlledProcessManagerServlet.java [hunk 3]
```diff
@@ -159,12 +161,29 @@ private ProcessInstance getProcessFromRequest(SlingHttpServletRequest request) {
...
-    private Map<String, Object> convertRequestMap(Map<String, String[]> parameterMap) {
-        return parameterMap.entrySet().stream()
+    private Map<String, Object> convertRequestMap(RequestParameterMap requestParameterMap) {
+        return requestParameterMap.entrySet().stream()
                 .filter(entry -> !ignoredInputs.contains(entry.getKey()))
                 .collect(Collectors.toMap(
                         entry -> entry.getKey(),
-                        entry -> (Object) (entry.getValue().length == 1 ? entry.getValue()[0] : entry.getValue())
+                        entry -> {
+                            final RequestParameter[] values = entry.getValue();
```
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/mcp/impl/ControlledProcessManagerServlet.java]: @badvision Changed this to handle file uploads.

File: bundle/src/main/java/com/adobe/acs/commons/mcp/form/FieldComponent.java (hunk 1 above)
Comment (badvision) [reply bundle/src/main/java/com/adobe/acs/commons/mcp/form/FieldComponent.java]: Yes, absolutely!  Thanks.  That was a future area of expansion and the next logical step for that bit.  I didn't like having to use a string array but the fact is that annotations provide you very few options so it was a necessary evil in order to pass less-structured data along.

File: bundle/src/main/java/com/adobe/acs/commons/mcp/impl/ControlledProcessManagerServlet.java (hunk 3 above)
Comment (badvision) [reply bundle/src/main/java/com/adobe/acs/commons/mcp/impl/ControlledProcessManagerServlet.java]: Oh, good thinking!  Just keep in mind it was designed such that single value = object, array => Array of objects, etc.  I haven't hooked up a component that generates multiple values as inputs into the servlet so this was an area of future expansion for multivalue fields, etc.

File: bundle/src/main/java/com/adobe/acs/commons/mcp/util/AnnotatedFieldDeserializer.java (hunk 2 above)
Comment (badvision) [reply bundle/src/main/java/com/adobe/acs/commons/mcp/util/AnnotatedFieldDeserializer.java]: Makes sense.

**Comment by badvision** [comment]
Looks great and I like that you added even more form components for folks to use!  Very cool.  Note that there's no need for NAME to be a static field on the process definition.  I was able to re-work things such that only the factory's getName() method has to return that value so it's not needed in more than one spot.

File: bundle/src/main/java/com/adobe/acs/commons/mcp/impl/ControlledProcessManagerServlet.java (hunk 3 above)
Comment (davidjgonzalez) [reply bundle/src/main/java/com/adobe/acs/commons/mcp/impl/ControlledProcessManagerServlet.java]: Not sure I follow this ...

File: content/src/main/content/jcr_root/apps/acs-commons/components/utilities/manage-controlled-processes/clientlibs/js/scriptRunner.js [hunk 4]
```diff
@@ -109,17 +109,20 @@ var ScriptRunner = {
         });
     },
     startProcess: function () {
-        var data = {};
-        jQuery("#processDefinitionInput form", window.top.document).serializeArray().map(function (x) {
-            data[x.name] = x.value;
-        });
+        /* Use FormData object to support file uploads */
```
Comment (davidjgonzalez) [inline content/src/main/content/jcr_root/apps/acs-commons/components/utilities/manage-controlled-processes/clientlibs/js/scriptRunner.js]: @badvision - just want to make sure you saw this change as well; I didn't quite grok why you were doing the map to the data object.

**Comment by badvision** [comment]
It all looks good to me but I don't have the final say to accept and merge the PR.

**Pull request merged**